libcst/codemod/commands/tests/test_remove_unused_imports.py

```diff
@@ -48,3 +48,32 @@ def test_type_annotations(self) -> None:
             x: a = 1
         """
         self.assertCodemod(before, before)
+
+    def test_dotted_imports(self) -> None:
+        before = """
+            import a.b, a.b.c
+            import e.f
+            import g.h
+            import x.y, x.y.z
+
+            def foo() -> None:
+                a.b
+                e.g
+                g.h.i
+                x.y.z
+        """
+
+        after = """
+            import a.b, a.b.c
```

Comment:
Why `a.b.c` is not removed?

Reply:
I understand a.b.c may have side effect and thus is not safe to remove. My question is how do we not remove it?
The ScopeVisitor looks like will use name `a.b` to store access to assignment `a.b`.
Is it because `import a.b.c` creates three assignment lookup entries (a, a.b, a.b.c), `a.b` access is recorded to the second one so, the import is not removed?
In the case of `x.y.z`, we only associate access `x.y.z` to `import x.y.z` instead of `import x.y` (which are x and x.y).

Reply:
It's because `a.b.c` is looked at in isolation (without considering the existence of the import for `a.b`).
So essentially because of the same reason this is not fixed:
```
import a
import a
a
```
This is a limitation of the `RemoveUnusedImportsVisitor`, it only removes unused imports, not duplicates.